L/LLVM/common.jl

```diff
@@ -0,0 +1,265 @@
+CMAKE_FLAGS+=(-DHAVE_HISTEDIT_H=Off)
+CMAKE_FLAGS+=(-DHAVE_LIBEDIT=Off)
+
+# We want a shared library
+CMAKE_FLAGS+=(-DLLVM_BUILD_LLVM_DYLIB:BOOL=ON)
+CMAKE_FLAGS+=(-DLLVM_LINK_LLVM_DYLIB:BOOL=ON)
+
+# Install things into $prefix, and make sure it knows we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
+
+# Julia expects the produced LLVM tools to be installed into tools and not bin
+# We can't simply move bin to tools since on MingW64 it will also contain the shlib.
+CMAKE_FLAGS+=(-DLLVM_TOOLS_INSTALL_DIR=${prefix}/tools)
+
+# Also build and install utils, since we want FileCheck, and lit
+CMAKE_FLAGS+=(-DLLVM_UTILS_INSTALL_DIR=${prefix}/tools)
+CMAKE_FLAGS+=(-DLLVM_INCLUDE_UTILS=True -DLLVM_INSTALL_UTILS=True)
+
+# Include perf/oprofile/vtune markers
+if [[ ${target} == *linux* ]]; then
+    CMAKE_FLAGS+=(-DUSE_PERF=1)
+    CMAKE_FLAGS+=(-DUSE_OPROFILE=1)
+fi
+if [[ ${target} == *linux* ]] || [[ ${target} == *mingw32* ]]; then
+    CMAKE_FLAGS+=(-DUSE_INTEL_JITEVENTS=1)
+fi
+
+# Tell LLVM where our pre-built tblgen tools are
+CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
+CMAKE_FLAGS+=(-DCLANG_TABLEGEN=${WORKSPACE}/bootstrap/bin/clang-tblgen)
+CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)
+
+# Explicitly use our cmake toolchain file
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=/opt/${target}/${target}.cmake)
+
+# Manually set the host triplet, as otherwise on some platforms it tries to guess using
+# `ld -v`, which is hilariously wrong.
+CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${target})
+
+# Tell LLVM which compiler target to use, because it loses track for some reason
+CMAKE_FLAGS+=(-DCMAKE_C_COMPILER_TARGET=${target})
+CMAKE_FLAGS+=(-DCMAKE_CXX_COMPILER_TARGET=${target})
+CMAKE_FLAGS+=(-DCMAKE_ASM_COMPILER_TARGET=${target})
+
+if [[ "${target}" == *apple* ]]; then
+    # On OSX, we need to override LLVM's looking around for our SDK
+    CMAKE_FLAGS+=(-DDARWIN_macosx_CACHED_SYSROOT:STRING=/opt/${target}/${target}/sys-root)
+
+    # LLVM actually won't build against 10.8, so we bump ourselves up slightly to 10.9
+    export MACOSX_DEPLOYMENT_TARGET=10.9
+    export LDFLAGS=-mmacosx-version-min=10.9
+
+    # We need to link against libc++ on OSX
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBCXX=ON)
+fi
+
+if [[ "${target}" == *apple* ]] || [[ "${target}" == *freebsd* ]]; then
+    # On clang-based platforms we need to override the check for ffs because it doesn't work with `clang`.
+    export ac_cv_have_decl___builtin_ffs=yes
+
+    # We don't use X-ray on BSD systems
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_XRAY=OFF)
+fi
+
+if [[ "${target}" == *mingw* ]]; then
+    CMAKE_CPP_FLAGS="${CMAKE_CPP_FLAGS} -remap -D__USING_SJLJ_EXCEPTIONS__ -D__CRT__NO_INLINE"
+    # Windows is case-insensitive and some dependencies take full advantage of that
+    echo "BaseTsd.h basetsd.h" >> /opt/${target}/${target}/include/header.gcc
```

Comment:
I had no idea this existed

Reply:
You just forgotten again, we talked about this when I originally added it...